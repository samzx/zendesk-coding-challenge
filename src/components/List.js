import React from "react";

import Listing from "./Listing";
import { listingsPerPage } from "../constants/constants";

class List extends React.Component {
  scrollToTop = () => {
    this._listDiv.scrollTop = 0;
  };

  generateList = tickets => {
    return tickets.map((listing, index) => (
      <Listing
        listing={listing}
        id={listing.id}
        index={index}
        key={`listing-${index}`}
        currentTicket={this.props.currentTicket}
        setTicket={this.props.setTicket}
        loading={this.props.loading}
      />
    ));
  };

  render() {
    return (
      <div className="list" ref={ref => (this._listDiv = ref)}>
        {this.props.data && this.generateList(this.props.data.tickets)}
      </div>
    );
  }
}

export default List;
