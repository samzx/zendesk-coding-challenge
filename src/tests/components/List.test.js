import React from "react";
import { shallow } from "enzyme";
import List from "../../components/List";

import threeTickets from "../fixtures/threeTickets";

describe("<List />", () => {
  it("should render List correctly", () => {
    const wrapper = shallow(<List />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should generate 3 Listings", () => {
    const wrapper = shallow(<List />);
    expect(wrapper.instance().generateList(threeTickets.tickets)).toHaveLength(
      3
    );
  });
});
