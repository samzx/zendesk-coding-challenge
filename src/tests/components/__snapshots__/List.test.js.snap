// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should render List correctly 1`] = `
<div
  className="list"
/>
`;
